### PR TITLE
Inject controller config into cloud

### PR DESCRIPTION
**Pull request by SimonRichardson**
The following adds the controller config to the cloud facade. This is a bit of a mess, as we have a lot of validation functions that are common, which look to me like they shouldn't be. A bit of duplication is ok, it would have been ok here.

## Checklist

- [x] Code style: imports ordered, good names, simple structure, etc
- [x] Comments saying why design decisions were made
- [x] Go unit tests, with comments saying what you're testing

## QA steps

```sh
$ juju bootstrap aws test --build-agent
```

## Links

**Jira card:** JUJU-4696

### CODE REVIEW

**Review by manadart — APPROVED** [review]
Another one down.

File: apiserver/facades/client/cloud/cloud_test.go [hunk 1]
```diff
@@ -1263,8 +1285,12 @@ func (s *cloudSuite) TestUpdateCredentialsAllModelsFailedValidationForce(c *gc.C
...
 
 	s.PatchValue(cloud.ValidateNewCredentialForModelFunc,
-		func(_ credentialcommon.PersistentBackend, _ context.ProviderCallContext,
-			_ names.CloudCredentialTag, _ *jujucloud.Credential, _ jujucloud.Cloud, migrating bool) (params.ErrorResults,
+		func(_ context.ProviderCallContext,
+			_ credentialcommon.PersistentBackend,
+			_ credentialcommon.ControllerConfigService,
+			_ names.CloudCredentialTag, _ *jujucloud.Credential,
+			_ jujucloud.Cloud,
+			_ bool) (params.ErrorResults,
```
[the remark this answers is not in the document]
Comment (SimonRichardson) [reply apiserver/facades/client/cloud/cloud_test.go:1293]: Except you do need slight different ones every so often.

**Pull request merged**